src/Tools/ExternalAccess/FSharp/Internal/Classification/FSharpClassificationService.cs

```diff
@@ -38,11 +39,15 @@ public void AddLexicalClassifications(SourceText text, TextSpan textSpan, Segmen
             result.AddRange(list);
         }
 
-        public async Task AddSemanticClassificationsAsync(Document document, TextSpan textSpan, ClassificationOptions options, SegmentedList<ClassifiedSpan> result, CancellationToken cancellationToken)
+        public async Task AddSemanticClassificationsAsync(Document document, TextSpan[] textSpans, ClassificationOptions options, SegmentedList<ClassifiedSpan> result, CancellationToken cancellationToken)
         {
-            using var _ = s_listPool.GetPooledObject(out var list);
-            await _service.AddSemanticClassificationsAsync(document, textSpan, list, cancellationToken).ConfigureAwait(false);
-            result.AddRange(list);
+            // F# does not support multiple text spans.
+            if (textSpans.Length == 1)
+            {
+                using var _ = s_listPool.GetPooledObject(out var list);
+                await _service.AddSemanticClassificationsAsync(document, textSpans[0], list, cancellationToken).ConfigureAwait(false);
+                result.AddRange(list);
+            }
```

Comment:
i would not do it this way.  just loop over each textspan and call into their service, aggregating their results into the result  array.

that way if roslyn ever decides to break up the classification call into small spans, F# will continue to work.